### ISSUE TITLE
The runner does not report the message of panics

### ISSUE THREAD

**Issue by xdoardo** [CHERIoT-Platform/cheri-rust#38]
Take, for example, the following test: 
```
#![no_std]

extern crate alloc;
extern crate cheriot;

#[no_mangle]
extern "C" fn test_panic() -> i32 {
    assert!(false, "hello!!!");
    0
}
```
when executing it the runner will correctly pick up the failure, but won't show the full message: 
```
[1/4] test panic ... fail
--- begin stdout from test runner --
Board file saved as  build/cheriot/cheriot/release/test.board.json
Remapping priority of thread 1 from 1 to 0
@rust-test-runner-sync-start

--- end stdout from test runner --
--- begin stderr from test runner --
tohost located at 0x80014fc8

--- end stderr from test runner --
running `xmake`
running cd "/var/folders/yk/spflkwdd62xd8zhy_nx9_6nr0000gn/T/cheri-runner/objs/unsafecell" && "xmake" "run"
running `xmake run`...ok
```

Instead running the firmware directly (without using the runner as intermediary) will print: 
```
$ xmake run
Board file saved as  build/cheriot/cheriot/release/test.board.json
Remapping priority of thread 1 from 1 to 0
Running file test.
ELF Entry @ 0x80000000
tohost located at 0x80014fc8
@rust-test-runner-sync-start
panicked at ./panic.rs:8:5:
hello!!!FAILURE: 1
error: execv(cheriot_sim test) failed(1)
warning: config.buildir() has been deprecated, please use config.builddir()
warning: add -v for getting more warnings ..
```

**Comment by xdoardo** [comment]
I think now we have the opposite problem! The runner now reports the message of panics twice if in tracing mode. 
@christiankissig, could you take a look? I think it should be enough to gate the second set of prints with a "if not in tracing mode.."
